dbt/adapters/snowflake/relation.py

```diff
@@ -0,0 +1,79 @@
+from dbt.adapters.default.relation import DefaultRelation
+from dbt.utils import filter_null_values
+
+
+class SnowflakeRelation(DefaultRelation):
+    DEFAULTS = {
+        'metadata': {
+            '_type': 'SnowflakeRelation'
+        },
+        'quote_character': '"',
+        'quote_policy': {
+            'database': False,
+            'schema': False,
+            'identifier': False,
+        },
+        'include_policy': {
+            'database': False,
+            'schema': True,
+            'identifier': True,
+        }
+    }
+
+    SCHEMA = {
+        'type': 'object',
+        'properties': {
+            'metadata': {
+                '_type': {
+                    'type': 'string',
+                    'const': 'SnowflakeRelation',
+                },
+            },
+            'type': {
+                'enum': DefaultRelation.RelationTypes + [None],
+            },
+            'path': DefaultRelation.PATH_SCHEMA,
+            'include_policy': DefaultRelation.POLICY_SCHEMA,
+            'quote_policy': DefaultRelation.POLICY_SCHEMA,
+            'quote_character': {'type': 'string'},
+        },
+        'required': ['metadata', 'type', 'path', 'include_policy',
+                     'quote_policy', 'quote_character']
+    }
+
+    @classmethod
+    def create_from_node(cls, profile, node, **kwargs):
+        return cls.create(
+            database=profile.get('database'),
+            schema=node.get('schema'),
+            identifier=node.get('name'),
+            **kwargs)
+
+    def matches(self, database=None, schema=None, identifier=None):
+        search = filter_null_values({
+            'database': database,
+            'schema': schema,
+            'identifier': identifier
+        })
+
+        if not search:
+            # nothing was passed in
+            pass
+
+        for k, v in search.items():
+            # snowflake upcases unquoted identiifers. so, when
+            # comparing unquoted identifiers, use case insensitive
+            # matching. when comparing quoted identifiers, use case
+            # sensitive matching.
+            if self.should_quote(k):
+                if self.get_path_part(k) != v:
+                    return False
+
+            else:
+                if self.get_path_part(k) != v.upper():
```

Comment:
should we uppercase both sides of this?

Reply:
i don't think so. one of the nice things about this is if the things was created with quotes previously, and the relation name is lowercased, then this will *not* match, which i think is the desired behavior.

it does pose a challenge for migrating existing snowflake projects to this version of dbt if the existing model names are all lowercased. we should chat about that. i think it's better to use this behavior though, otherwise i think dbt may behave inconsistently

Reply:
roger, sounds good to me